### PR TITLE
Consider new control-plane label

### PR DESCRIPTION
**Pull request by mogottsch**
This PR considers the new label `node-role.kubernetes.io/control-plane` when determining whether a node is a control-plane/master node.

The lines I changed are not covered by any tests and I don't know how to test the operator manually.
Could you tell me the necessary steps to take in order to test this?

Towards https://github.com/giantswarm/roadmap/issues/2471

### CODE REVIEW

File: pkg/encryption/encryption.go [hunk 1]
```diff
@@ -518,24 +518,29 @@ func (s *Service) areAllMasterNodesUsingLatestConfig(ctx context.Context, wcClie
...
-		ctrlclient.MatchingLabels{key.MasterNodeLabel: ""})
-	if err != nil {
-		return false, microerror.Mask(err)
+	// get all master nodeItems
+	nodeItems := []v1.Node{}
+	for _, label := range key.MasterNodeLabels {
+		var tmpNodes v1.NodeList
+		err = wcClient.List(ctx,
+			&tmpNodes,
+			ctrlclient.MatchingLabels{label: ""},
```
Comment (whites11) [inline pkg/encryption/encryption.go:527]: I guess the `""` should be `label` instead?

Comment (mogottsch) [reply pkg/encryption/encryption.go:527]: I don't think so.

`MatchingLabels` is a `map[string]string`, which maps label keys to label values.
The controlplane nodes have the key-value-pair `node-role.kubernetes.io/control-plane: ""` in  `metadata.labels[]`.

Or am I missing something here?

Note, that before my change it also was:
```
		ctrlclient.MatchingLabels{key.MasterNodeLabel: ""})
```

Comment (whites11) [reply pkg/encryption/encryption.go:527]: yeah right, dunno what I saw

**Comment by whites11** [comment]
> @whites11 Do you think it's fine to merge this, without testing?

The answer to this question has always been and will always be no.
If you can't or are unable to test it assign to team phoenix and we'll eventually take care of testing and merging

**Comment by mogottsch** [comment]
I don't really know what this operator does, so I don't think I am able to test this. I assigned you.

If I can be of any help here, just let me know.

**Comment by whites11** [comment]
added to team board for prioritization

**Comment by bdehri** [comment]
@whites11 created and deleted a couple of clusters without issues. I can see that encryption provider secret was created successfully. Shall we merge this one?

**Comment by bdehri** [comment]
@whites11 created and deleted clusters successfully, also I can see that encryption provider secret was created successfully according to logs. Shall we merge this one?

**Comment by calvix** [comment]
this is actually used during key rotation only so its not used during cluster creation at all

**Pull request merged**